plugin/pkg/scheduler/factory/factory.go

```diff
@@ -227,9 +237,7 @@ func (f *ConfigFactory) CreateFromKeys(predicateKeys, priorityKeys sets.String,
 		Algorithm:  algo,
 		Binder:     &binder{f.Client},
 		NextPod: func() *api.Pod {
-			pod := f.PodQueue.Pop().(*api.Pod)
-			glog.V(2).Infof("About to try and schedule pod %v", pod.Name)
-			return pod
+			return f.getNextPod()
```

Comment:
Changing to NextPod: f.getNextPod() would require many follow-up changes which I think is not necessary. It is not as easy as I thought. Other than that, all comments are respected in the updated PR, including an integration test.